packages/maskbook/src/plugins/Collectible/SNSAdaptor/Collectible.tsx

```diff
@@ -301,11 +304,11 @@ export function Collectible(props: CollectibleProps) {
                     </div>
                 </CardActions>
             </CollectibleCard>
-            {asset.value?.end_time && isValidDate(asset.value.end_time) && (
+            {isEndTimeVisible && (
                 <Box sx={{ marginTop: 1 }}>
                     <Typography className={classes.countdown}>
                         {t('plugin_collectible_sale_end', {
-                            time: formatDateTime(asset.value.end_time, 'yyyy-MM-dd HH:mm:ss'),
+                            time: formatDateTime(asset.value?.end_time ?? Date.now(), 'yyyy-MM-dd HH:mm:ss'),
```

Comment:
I think we can assert `end_time` is non-null here.


```suggestion
                            time: formatDateTime(asset.value!.end_time!, 'yyyy-MM-dd HH:mm:ss'),
```

I thought ts@4.4 could understand this, but [it didn't](https://www.typescriptlang.org/play?ts=4.4.0-beta&ssl=13&ssc=1&pln=13&pc=2#code/C4TwDgpgBAggznCAnYUDeBYAUFXUBuAhgDYCuEA-AFzrZ71QQB2AJtVHMEgJZMDmdPAF9sIrNgDGAeyacohBMmA14iFFAC86MZJlyAFgoCirTVACE5hWuAA6ImUq3mLbNm4AzKAApDcEywAlLQ4eNKyUsQQtsRSfN7WSvYk5M6sgaJAA). Do you have any idea? @Jack-Works

Reply:
What problem do you encountered?

Reply:
`isEndTimeVisible` already checks `asset.value.end_time` above, then I thought ts@4.4 could understand `asset.value.end_time` is non-null here.